src/mcnp/mcnp_funcs.cpp

```diff
@@ -317,16 +319,18 @@ void write_surface_cards(std::ostringstream& lcadfile) {
 
 // write out all the tally data from the uwuw file
 void write_material_data(std::ostringstream& lcadfile) {
-  std::map<std::string, pyne::Material> material_library = workflow_data->material_library;
+  pyne::MaterialLibrary material_library = workflow_data->material_library;
   // loop over all tallies
   std::cout << "Writing Materials ..." << std::endl;
 
   lcadfile << "C materials from library" << std::endl;
   // loop over the material and print them out
-  for (std::map<std::string, pyne::Material>::const_iterator it = material_library.begin() ;
-       it != material_library.end() ; ++it) {
-    pyne::Material new_material = (it->second);
-    std::string material_card = new_material.mcnp();
+
+  pyne::MaterialLibrary::iterator it;
+
+  for (it = material_library.begin(); it != material_library.end() ; ++it) {
+    std::shared_ptr<pyne::Material> new_material = (it->second);
+    std::string material_card = new_material->mcnp();
     lcadfile << material_card;
```

Comment:
```suggestion
    lcadfile << (it->second)->material_card;
```

Reply:
do you mean ?
`lcadfile << (it->second)->mcnp();`

Reply:
Haha yes. Sorry. Editing.